packages/core-plugin-api/src/plugin/types.ts

```diff
@@ -88,6 +145,7 @@ export type PluginConfig<
   routes?: Routes;
   externalRoutes?: ExternalRoutes;
   featureFlags?: PluginFeatureFlagConfig[];
+  info?: PluginConfigInfo;
```

Comment:
This might seem a bit silly, but I'm thinking it's worth making the `package.json`s lazy-loaded or in some other way make sure we're not eagerly loading in too much data. Considering for example the `package.json` of the catalog, it's roughly 2000 bytes. Make that 100 plugins and you've got 200kB added to the main bundle.

Reply:
Yeah, I just `cat`ed **all** the package.jsons in plugins/ and packages/ now, it's ~250kB although gzipped down to 17kB. Which is really not that much, considering..

Lazy-loading is an interesting idea, but it makes other things complicated, like, if you want to use this data in code, to e.g. log (as with #10259) or if you want to view all plugins (#10260) or if you want ErrorBoundaries, Analytics and other things to take advantage of this metadata, like ownership (based on package.json `name`), all of this would have to be async.

I'm really not sure what'd be the right procedure. Maybe a webpack filter for `package.json` imports that cuts off all unneccesary things (*dependencies) or only filters what we need. That'd trim the final size. Perhaps something like an import loader a la `'plugin-metadata!../package.json'`

Reply:
resolved by allowing lazy loaded (dynamically imported) `package.json`